ddtrace/tracer/sampler.go

```
@@ -188,19 +188,76 @@ func newRulesSampler(rules []SamplingRule) *rulesSampler {
+// and combines them with spanSamplingRules
 func samplingRulesFromEnv() ([]SamplingRule, error) {
+	errs := []string{}
+	spanRules, err := spanSamplingRulesFromEnv()
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	traceRules, err := traceSamplingRulesFromEnv()
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	rules := []SamplingRule{}
+	if traceRules != nil {
+		rules = traceRules
+	}
+	if spanRules != nil {
+		rules = append(rules, spanRules...)
+	}
+	if len(errs) != 0 {
+		return rules, fmt.Errorf("%s", strings.Join(errs, "\n\t"))
+	}
+	return rules, nil
+}
+
+// traceSamplingRulesFromEnv parses sampling rules from the DD_TRACE_SAMPLING_RULES environment variable.
+func traceSamplingRulesFromEnv() ([]SamplingRule, error) {
 	rulesFromEnv := os.Getenv("DD_TRACE_SAMPLING_RULES")
 	if rulesFromEnv == "" {
 		return nil, nil
 	}
-	jsonRules := []struct {
-		Service string      `json:"service"`
-		Name    string      `json:"name"`
-		Rate    json.Number `json:"sample_rate"`
-	}{}
-	err := json.Unmarshal([]byte(rulesFromEnv), &jsonRules)
+	return processSamplingRulesFromEnv([]byte(rulesFromEnv), TraceSamplingRuleType)
+}
+
+// spanSamplingRulesFromEnv parses sampling rules from the DD_SPAN_SAMPLING_RULES and
+// DD_SPAN_SAMPLING_RULES_FILE environment variables.
+func spanSamplingRulesFromEnv() ([]SamplingRule, error) {
+	errs := []string{}
+	rules, err := processSamplingRulesFromEnv([]byte(os.Getenv("DD_SPAN_SAMPLING_RULES")), SingleSpanSamplingType)
+	if err != nil {
+		errs = append(errs, err.Error())
+	}
+	if len(rules) != 0 {
+		if os.Getenv("DD_SPAN_SAMPLING_RULES_FILE") != "" {
+			log.Warn("DIAGNOSTICS Error(s): DD_SPAN_SAMPLING_RULES is available and will take precedence over DD_SPAN_SAMPLING_RULES_FILE")
+		}
+		return rules, err
+	}
+	rulesFile := os.Getenv("DD_SPAN_SAMPLING_RULES_FILE")
+	if rulesFile == "" {
+		return rules, err
+	}
+	rulesFromEnvFile, err := os.ReadFile(rulesFile)
+	if err != nil {
```

If this returned an error, `rulesFromEnvFile` is just going to be empty, so passing it to `processSamplingRulesFromEnv` isn't going to be useful here.

What I imagine we're trying to do here is to gather the rules, either from the env or from a file, then pass those rules to `processSamplingRulesFromEnv`. Before that, if any errors were found during this function, we wouldn't have any useful rules to pass to that function, so we should probably just return that error, right?